scripts/tests/sanitycheck/test_reporting_testsuite.py

```diff
@@ -0,0 +1,151 @@
+    inst2 = class_testsuite.instances.get("demo_board_2/scripts/tests/sanitycheck/test_data/testcases/tests/test_a/test_a.check_2")
+    inst1.status = "failed"
+    inst2.status = "skipped"
+    filename = test_data + "sanitycheck.xml"
+    class_testsuite.xunit_report(filename)
+    assert os.path.exists(filename)
+
+    filesize = os.path.getsize(filename)
+    assert filesize != 0
+
+    fails, passes, errors, skips = 0, 0, 0, 0
+    for instance in class_testsuite.instances.values():
+        if instance.status in ["failed", "timeout"]:
+            if instance.reason in ['build_error', 'handler_crash']:
+                errors += 1
+            else:
+                fails += 1
+        elif instance.status == 'skipped':
+            skips += 1
+        else:
+            passes += 1
+
+    tree = ET.parse(filename)
+    assert int(tree.findall('testsuite')[0].attrib["skip"]) == int(skips)
+    assert int(tree.findall('testsuite')[0].attrib["failures"]) == int(fails)
+    assert int(tree.findall('testsuite')[0].attrib["errors"]) == int(errors)
+    assert int(tree.findall('testsuite')[0].attrib["tests"]) == int(passes+fails+skips+errors)
+
+    for index in range(0, len(class_testsuite.instances)):
+        if len(list(tree.findall('testsuite')[0][index])) != 0:
+            if tree.findall('testsuite')[0][index][0].attrib["type"] == "failure":
+                assert tree.findall('testsuite')[0][index].attrib["classname"] == \
+                       (inst1.platform.name + ':' + inst1.testcase.name)
+            elif tree.findall('testsuite')[0][index][0].attrib["type"] == "skipped":
+                assert tree.findall('testsuite')[0][index].attrib["classname"] == \
+					   (inst2.platform.name + ':' + inst2.testcase.name)
+    os.remove(filename)
+
+def test_compare_metrics(class_testsuite, test_data, instances_fixture, caplog):
+    """ Testing compare_metrics function of Testsuite class in sanitycheck
+	Test 1: Error message is said if file sanitycheck.csv file doesnt exist
+	Test 2: Assert on compare_metrics results for expected values"""
+    class_testsuite.instances = instances_fixture
+    for instance in class_testsuite.instances.values():
+        instance.metrics["ram_size"] = 5
+        instance.metrics["rom_size"] = 9
+    filename_not_exist = test_data + "sanitycheck1.csv"
+    class_testsuite.compare_metrics(filename_not_exist)
+    assert "Cannot compare metrics, " + filename_not_exist + " not found" in caplog.text
+
+    filename = test_data + "sanitycheck.csv"
+    results = class_testsuite.compare_metrics(filename)
+    for instance in class_testsuite.instances.values():
+        for res in results:
+            if (res[0].platform.name == instance.platform.name) and \
+               (res[0].testcase.name == instance.testcase.name):
+                assert res[0].platform.name == instance.platform.name
```

Comment:
your`if` is ensuring that `res[0].platform.name == instance.platform.name` isn't ? This assert is useless.

Reply:
Have put the assert statement before this if expression now.